patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -5047,10 +5080,29 @@
 +		/*
  		if (tile.type == 4 && tile.type == createTile)
  			return tile.frameY != num * 22;
-+		*/
  
  		if (tile.type == 215 && tile.type == createTile)
  			return tile.frameX / 54 != num;
++		*/
++
++		if (TileID.Sets.Campfire[tile.type] && TileID.Sets.Campfire[createTile]) {
++			if (tile.frameX / 54 == num)
++				return tile.type != createTile;
++
++			return true;
```

Comment:
`return tile.type != createTile || tile.frameX / 54 != num`

Reply:
Might be nice to edit the torch patch above with the same `||` formatting, and split the commented vanilla code into 2 groups  (torch and campfire), each preceeding the modded variant.